Add native token how-tos (#541)
iotaledger/iota-sdk: documentation/sdk/docs/how_tos/native_tokens/destroy_foundry.mdx

---
title: Destroy a Foundry
sidebar_label: Destroy a Foundry

> description: 'How to destroy a foundry using wallet.rs.'

Reviewer: ```suggestion
description: 'How to destroy a foundry using iota-sdk'
```

image: /img/logo/iota_mark_light.png
keywords:
- how to
- destroy
- destroy foundry
- foundry
- nodejs
- python
- rust
---

import CodeBlock from '@theme/CodeBlock';
import Tabs from "@theme/Tabs";
import TabItem from "@theme/TabItem";
import NodejsCode from '!!raw-loader!../../../../../bindings/nodejs/examples/how_tos/native_tokens/destroy_foundry.ts';
import PythonCode from '!!raw-loader!../../../../../bindings/python/examples/how_tos/native_tokens/destroy_foundry.py';
import RustCode from '!!raw-loader!../../../../../sdk/examples/how_tos/native_tokens/destroy_foundry.rs';

You can destroy a foundry by calling the `Account.burn(burn, options)` function. The function will
destroy a foundry output as long as its circulating

> [native token](https://wiki.iota.org/shimmer/introduction/explanations/ledger/foundry) supply is zero. Any native tokens

Reviewer: Maybe also mention that this only works if no native tokens have been burned?

Author: Hmm. Maybe we should add an admonition for that. Let me do that

Author: Wdyt? https://github.com/iotaledger/iota-sdk/pull/541/commits/c7a9f68f667ba478ad38a1eab05223b9f5e5f855

Reviewer: Looks good

which were minted by other foundries will be sent to the controlling

> [alias](https://wiki.iota.org/shimmer/introduction/explanations/ledger/alias).

Reviewer: > Any native tokens
which were minted by other foundries will be sent to the controlling [alias]

This isn't happening

Author: Who wrote it than? 😅 

```suggestion
[native token](https://wiki.iota.org/shimmer/introduction/explanations/ledger/foundry) supply is zero. 
```

## Code Example

Before you run the example you should make sure the first foundy on your account has zero circulating supply. If you have no available
foundries, you can create one by [minting a native token](mint.mdx). If you've already minted your
tokens but need to empty the foundry, you can decrease your native token supply by [melting them](decrease_supply.mdx).

The following example will:

> 1. Create an account manager.

Reviewer: No

Author: ```suggestion
1. Create a wallet.
```
